pagopa/io-app
# chore: [IOCOM-1741] Notification engament modal - Phase 1

## Short description
This PR adds the engagement modal for users that have disabled  the system push notification permission.
| Onboarding<br/>No modal|Onboarding<br />Modal|Biometric<br/>Modal|
|-|-|-|
|<video src="https://github.com/user-attachments/assets/63f576f8-7039-4e03-8a18-590870aaf681"/>|<video src="https://github.com/user-attachments/assets/9ec31022-b025-4ab8-9956-1d3737fd2366" />|<video src="https://github.com/user-attachments/assets/914f571e-cc19-40df-8458-95fb0c2ee897" />|

## List of changes proposed in this pull request
- `SystemNotificationPermissionsScreen`: UI/UX. Dispatches an `setEngagementScreenShown` action upon first rendering. In order to be shown as a modal, it has been added to the `AuthenticatedStackNavigator`
- `useEngagementScreen` hook: calls `shouldShowEngagementScreenSelector` and displays the screen above if the output is `true`
- `MessagesHomeScreen`: calls the hook above
- `shouldShowEngagementScreenSelector`: outputs true when (1) the system notification permissions are disabled, (2) the startup saga has finished running, (3) `SystemNotificationPermissionsScreen` has never been shown and (4) the user does not come from the onboarding push notification instructions.
  - (1): runtime data saved into `notifications.environment.systemNotificationsEnabled`, updated upon receiving `updateSystemNotificationsEnabled` action, which is dispatched either by `profileAndSystemNotificationsPermissions` or `checkNotificationPermissionsOnAppForegroundState` sagas
  - (2): runtime data saved into `notifications.environment.applicationInitialized`, dispatched at the end of the `startupSaga` with `applicationInitialized` action
  - (3): persisted data saved into `notifications.userBehaviour.engagementScreenShown`, dispatched by `SystemNotificationPermissionsScreen` with `setEngagementScreenShown` action
  - (4): runtime data saved into `notifications.environment.onboardingInstructionsShown`, dispatched by `OnboardingNotificationsInfoScreenConsent` with `notificationsInfoScreenConsent` action


- `permissions` reducer renamed in `environment` reducer

## How to test
Using the io-dev-api-server, on a real device, check that:
- During a first time login, with disabled push notifications, the modal is not shown if the user has seen the onboarding instructions
- The same as above but, if the user has not seen the onboarding instructions, then the modal is  shown
- If the user is logged-in and has yet to see the modal, if the application is started from scratch, after the biometric identification, the modal is shown
- The modal is never shown again unless the user does a logout and the login is performed by a different user
